### ISSUE TITLE
Soroban - binascii.Error: Non-hexadecimal digit found

### ISSUE THREAD

**Issue by yuriescl** [StellarCN/py-stellar-base#744]
Getting this error when invoking a Soroban contract deployed today (2023-July-07):
```
Traceback (most recent call last):
  File "/home/yuri/test.py", line 12, in <module>
    .append_invoke_contract_function_op(
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/yuri/.cache/pypoetry/virtualenvs/test-YsNgcThI-py3.11/lib/python3.11/site-packages/stellar_sdk/transaction_builder.py", line 1280, in append_invoke_contract_function_op
    stellar_xdr.SCBytes(binascii.unhexlify(contract_id))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
binascii.Error: Non-hexadecimal digit found
```

SDK version:
`stellar-sdk @ git+https://github.com/StellarCN/py-stellar-base.git@1960f2f227fba8fe4a2215116d6713134c894645`

## How to reproduce
Run below Python script:
```
from stellar_sdk import TransactionBuilder
from stellar_sdk.soroban.server import SorobanServer
server = SorobanServer("https://rpc-futurenet.stellar.org:443/")
tx = (
    TransactionBuilder(
        server.load_account("GAQJXP4ERQVKN3WCLUUXBVJXHJMPLR2JTETPJC3GLXSEVVQOZAUPA3KW"),
        "Test SDF Future Network ; October 2022",
        base_fee=300000,
    )
    .set_timeout(30)
    .append_invoke_contract_function_op(
        "CCEYDXCRHLPTNASWIA3YHFANN5MCZNDEM63CKKVWCRPOKGO53PUBFNQY",
        "admin",
        [],
        auth=None,  # type: ignore
    )
    .build()
)
```
It appears that the Python SDK is not yet up-to-date with latest contract id format.

**Comment by overcat** [comment]
Sorry for the inconvenience caused. Currently please use the following code to retrieve the contract ID:

```
from stellar_sdk import StrKey
StrKey.decode_contract('CCEYDXCRHLPTNASWIA3YHFANN5MCZNDEM63CKKVWCRPOKGO53PUBFNQY').hex()
```

**Comment by overcat** [comment]
In the newly released 9.0.0a0, we have added support for `C...` key and removed support for hex.
